File: app/src/at-vinhnguyen/java/asiantech/internship/summer/javacore/exercise2/BaseNumber.java
package asiantech.internship.summer.javacore.exercise2;

import android.util.Log;
import java.util.Scanner;

public class BaseNumber {

    private static final String TAG = BaseNumber.class.getSimpleName();

    public static void main(String[] args) {
        Scanner sn = new Scanner(System.in, "UTF-8");
        Log.d(TAG, "Nhap he co so ban can chuyen: ");
        int base = Integer.parseInt(sn.nextLine());
        Log.d(TAG, "Nhap so o he thap phan ban can chuyen: ");
        int number = Integer.parseInt(sn.nextLine());
        Log.d(TAG, "So sau khi chuyen: " + convertBaseNumber(base, number));
    }

    private static int convertBaseNumber(int baseNumber, int number) {
        int i = 1;
        int result = 0;
        while (number > 1) {
            result += number % baseNumber * i;
            number /= baseNumber;
            i *= 10;
        }
        return result;
    }
}

File: app/src/at-vinhnguyen/java/asiantech/internship/summer/kotlincore/exercise2/BaseNumber.kt
package asiantech.internship.summer.kotlincore.exercise2

import asiantech.internship.summer.kotlincore.inputIntNumber

object BaseNumber {

    @JvmStatic
    fun main(args: Array<String>) {
        println("Nhap he co so ban can chuyen: ")
        val coSo = inputIntNumber()
        println("Nhap so o he thap phan ban can chuyen: ")
        val so = inputIntNumber()
        println("So sau khi chuyen: " + convertBaseNumber(
                coSo, so))
    }

    private fun convertBaseNumber(baseNumber: Int, number: Int): Int {
        var a = number
        var i = 1
        var result = 0
        while (a > 1) {
            result += a % baseNumber * i
            a /= baseNumber
            i *= 10
        }
        return result
    }
}
